cli/azd/pkg/containerapps/container_app.go

```diff
@@ -116,107 +159,166 @@ func (cas *containerAppService) AddRevision(
+// This method will block until the revision is ready, reaches a terminal state or times out
+// Expects that revision to be in the following states:
+// ProvisioningState: "Provisioned"
+// HealthState: "Healthy"
+func (cas *containerAppService) ValidateRevision(
 	ctx context.Context,
 	subscriptionId string,
 	resourceGroupName string,
 	appName string,
-	containerApp *armappcontainers.ContainerApp,
-) (*armappcontainers.ContainerApp, error) {
-	// If the container app doesn't have any secrets, we don't need to do anything
-	if len(containerApp.Properties.Configuration.Secrets) == 0 {
-		return containerApp, nil
-	}
-
-	appClient, err := cas.createContainerAppsClient(ctx, subscriptionId)
+	revisionName string,
+) (*armappcontainers.Revision, error) {
+	revisionsClient, err := cas.createRevisionsClient(ctx, subscriptionId)
 	if err != nil {
 		return nil, err
 	}
 
-	// Copy the secret configuration from the current version
-	// Secret values are not returned by the API, so we need to get them separately
-	// to ensure the update call succeeds
-	secretsResponse, err := appClient.ListSecrets(ctx, resourceGroupName, appName, nil)
-	if err != nil {
-		return nil, fmt.Errorf("listing secrets: %w", err)
-	}
+	var result *armappcontainers.Revision
 
-	secrets := []*armappcontainers.Secret{}
-	for _, secret := range secretsResponse.SecretsCollection.Value {
-		secrets = append(secrets, &armappcontainers.Secret{
-			Name:  secret.Name,
-			Value: secret.Value,
-		})
-	}
+	// Wait 10 seconds between retries
+	backOffDuration := retry.NewConstant(time.Second * 10)
 
-	containerApp.Properties.Configuration.Secrets = secrets
+	// Keep trying for up to 10 minutes or has reached a terminal state
+	maxDuration := retry.WithMaxDuration(time.Minute*10, backOffDuration)
+
+	err = retry.Do(ctx, maxDuration, func(ctx context.Context) error {
+		getRevisionResponse, err := revisionsClient.GetRevision(ctx, resourceGroupName, appName, revisionName, nil)
+		if err != nil {
+			return fmt.Errorf("getting revision '%s': %w", revisionName, err)
+		}
+
+		revision := getRevisionResponse.Revision
+
+		isProvisioned := false
+		isHealthy := false
+
+		switch *revision.Properties.ProvisioningState {
+		case armappcontainers.RevisionProvisioningStateProvisioned:
+			isProvisioned = true
+		case armappcontainers.RevisionProvisioningStateProvisioning:
+			return retry.RetryableError(fmt.Errorf("revision '%s' is not ready", revisionName))
+		case armappcontainers.RevisionProvisioningStateFailed:
+			provisioningError := revision.Properties.ProvisioningError
+			if provisioningError == nil {
+				provisioningError = convert.RefOf("unknown error")
+			}
+			return fmt.Errorf("revision '%s' failed to provision, %s", revisionName, *provisioningError)
+		}
+
+		switch *revision.Properties.HealthState {
+		case armappcontainers.RevisionHealthStateHealthy:
+			isHealthy = true
+		case armappcontainers.RevisionHealthStateUnhealthy, armappcontainers.RevisionHealthStateNone:
```

Comment:
We're treating health state = None the same as health state = Unhealthy. Is this right?

Reply:
Right now there isn't any description what `HealthState=None` means. 
Based on the [docs](https://learn.microsoft.com/en-us/azure/container-apps/health-probes?tabs=arm-template) default health probes are added for containers with ingress enabled.

The only thing I would be able to infer from the docs is containers that do not accept ingress and have not specified their own health probes may be reported as a `None` state.  I'll see if I can do a quick test to confirm.

Reply:
Unable to get a container into the `None` health state.  I am OK on waiting for `Healthy` status until this state is better known.